src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -243,5 +247,21 @@ internal static ValueTask WriteAtOffsetAsync(SafeFileHandle handle, ReadOnlyMemo
 
         private static ValueTask WriteGatherAtOffsetAsync(SafeFileHandle handle, IReadOnlyList<ReadOnlyMemory<byte>> buffers, long fileOffset, CancellationToken cancellationToken)
             => handle.GetThreadPoolValueTaskSource().QueueWriteGather(buffers, fileOffset, cancellationToken);
+
+        private static bool NeedsNonOffsetFallback(SafeFileHandle handle)
+        {
+            // We need to fallback to the non-offset version for certain file types
+            // e.g: character devices (such as /dev/tty), pipes, and sockets.
+            Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
```

Comment:
Would it be easy for a future refactor to introduce something that resets last error before this is called? Maybe last error should be passed in so it's visible.